src/fdtdx_studio/ui/popups/pop_up_new_source.py

```
@@ -270,23 +288,28 @@ def make_source_mode_options(self):
     self.wave_button.close()
     self.wave = wave
     self.wave_button.text = f'{wave}'
 
   def set_direction(self, direction):
     """Sets the direction and updates the button display."""
+    assert self.direction_button
     self.direction_button.close()
     self.input_direction = direction
     self.direction_button.text = f'Direction: {direction}'
 
   def set_filter(self, filter):
```

Ruff (0.15.10)

[error] Function argument `filter` is shadowing a Python builtin

(A002)